File: iosApp/iosAppTests/offer/TestSwapTruthSource.swift
//
//  TestSwapTruthSource.swift
//  iosAppTests
//
//  Created by jimmyt on 8/15/22.
//  Copyright © 2022 orgName. All rights reserved.
//

import Foundation
@testable import iosApp

/**
 A basic `SwapTruthSource` implementation for testing.
 */
class TestSwapTruthSource: SwapTruthSource {
    var swaps: [UUID : Swap] = [:]
}

File: androidApp/src/test/java/com/commuto/interfacemobile/android/offer/TestSwapTruthSource.kt
package com.commuto.interfacemobile.android.offer

import androidx.compose.runtime.mutableStateMapOf
import androidx.compose.runtime.snapshots.SnapshotStateMap
import com.commuto.interfacemobile.android.swap.Swap
import com.commuto.interfacemobile.android.swap.SwapTruthSource
import java.util.*

/**
 * A basic [SwapTruthSource] implementation for testing.
 */
class TestSwapTruthSource: SwapTruthSource {
    override var swaps: SnapshotStateMap<UUID, Swap> = mutableStateMapOf()
    override fun addSwap(swap: Swap) {
        swaps[swap.id] = swap
    }
}